Automattic/wp-calypso
# browser-data-collector: export API types

`@automattic/browser-data-collector` exports two main methods, `start` and `stop`. These methods optionally take an array of `Collector` objects.

While the package exports types for the methods, these types don't include a definition for `Collector` nor the types it depends on. This PR exports those types to allow consumers to use them in their collector implementations.

CC @gabrielcaires

## Proposed Changes

* Add `Collector` and various `Report`-related types to type exports.
* Switch from `types.d.ts` to a proper `types.ts` that is explicitly imported throughout the package.
* Clean up types a bit to extend globals instead of replacing them.

## Why are these changes being made?

* To improve types in consumers of `@automattic/browser-data-collector`

## Testing Instructions

These changes are all type-related, so it should suffice to ensure that no additional type warnings or errors were created.

## Pre-merge Checklist

N/A
